### PR TITLE
authz: merge `PermsFetcher` into `Provider` interface

### PR DESCRIPTION
**Pull request by unknwon**
We now migrated all authz providers to support fetch permissions in user- and repo-centric, it's time to merge the interface.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -859,6 +859,7 @@ github.com/sourcegraph/syntaxhighlight v0.0.0-20170531221838-bd320f5d308e h1:qpG
 github.com/sourcegraph/syntaxhighlight v0.0.0-20170531221838-bd320f5d308e/go.mod h1:HuIsMU8RRBOtsCgI77wP899iHVBQpCmg4ErYMZB+2IA=
 github.com/sourcegraph/zoekt v0.0.0-20200318141102-0b140b7dc6c9 h1:vUz68uAPzIPdz9ibYgEOvRh33fErNN4LFYnGSCObP28=
 github.com/sourcegraph/zoekt v0.0.0-20200318141102-0b140b7dc6c9/go.mod h1:myt55/e315SlJX+bvPR+Y1wnOijZ+tcoG0gkjexi9F4=
+github.com/sourcegraph/zoekt v0.0.0-20200401202556-ef3ec2393b46 h1:4bfwcMwJFWRK/BHzW0yXGy0XGTXGWYs9WwdAWR5dKOQ=
```
Comment (unknwon) [inline go.sum:862]: This is a side-effect after rebased from `master`.

File: cmd/frontend/authz/iface.go [hunk 2]
```diff
@@ -55,6 +55,26 @@ type Provider interface {
...
+
+	// FetchRepoPerms returns a list of user IDs (on code host) who have read access to
+	// the given repository/project on the code host. The user ID should be the same value
+	// as it would be used as extsvc.Account.AccountID. The returned list should
+	// include both direct access and inherited from the group/organization/team membership.
+	//
+	// Because permissions fetching APIs are often expensive, the implementation should
+	// try to return partial but valid results in case of error, and it is up to callers
+	// to decide whether to discard.
+	FetchRepoPerms(ctx context.Context, repo *extsvc.Repository) ([]extsvc.ExternalAccountID, error)
```
Comment (tsenart) [inline cmd/frontend/authz/iface.go:76]: Since we renamed `extsvc.ExternalAccount` to `extsvc.Account`, should we do the same for these ID types?

Comment (unknwon) [reply cmd/frontend/authz/iface.go:76]: Thought about it too, will do.

Comment (unknwon) [reply cmd/frontend/authz/iface.go:76]: PR sent: https://github.com/sourcegraph/sourcegraph/pull/9512

**Pull request merged**